buildSrc/private/src/main/kotlin/androidx/build/AndroidXComposeMultiplatformExtensionImpl.kt

```diff
@@ -433,4 +436,41 @@ private fun Project.publishAndroidxReference(target: KotlinAndroidTarget) {
             }
         }
     }
-}
\ No newline at end of file
+}
+
+/**
+ * K/Native stores the dependencies in klib manifest and tries to resolve them during compilation.
+ * Since we use project dependency - implementation(project(...)), the klib manifest will reference
+ * our groupId (for example org.jetbrains.compose.collection-internal instead of androidx.collection).
+ * Therefore, the dependency can't be resolved since we don't publish libs for some k/native targets.
+ *
+ * To fix that, we need to make sure
+ * that the project dependency is substituted by a module dependency (from androidx).
+ * We do this here. It should be called only for appropriate k/native targets.
+ *
+ * For available androidx targets see:
+ * https://maven.google.com/web/index.html#androidx.annotation
+ * https://maven.google.com/web/index.html#androidx.collection
+ */
+private fun KotlinNativeTarget.substituteForOelPublishedDependencies() {
+    val comp = compilations.getByName("main")
+    val androidAnnotationVersion = project.findProperty("oel.androidx.annotation.version")!!
+    val androidCollectionVersion = project.findProperty("oel.androidx.collection.version")!!
```

Comment:
As mentioned, it doesn't seems as generic solution, and it's not clear how it can be reused in next modules (we already have a few).
Also, I'd like to put TODO about moving it out of Compose plugin here too - both of these libraries should NOT include compose plugin in their configs

Reply:
What other modules do you have in mind so I can understand the usecase better?
What targets for them do you expect to be published by androidx and what targets to publish on our side?

Reply:
This specific substitution is required for each library published by androidx and used in compose libs as dependency.
For every new such lib, a new couple of lines will need to be added here (only if it concerns native targets)

Reply:
> What other modules do you have in mind

For now `lifecycle-common` and `lifecycle-runtime` (+ `-ktx`). A few others on the way: `savedstate`, other `lifecycle-*`, it was an idea to republish `paging-compose` with ref to Compose Multiplatform. A  bit later we'll talk about `navigation-*` set of libraries too.

Based on such count, I don't like hardcoding anything like that in the plugin 

> What targets for them do you expect to be published by androidx and what targets to publish on our side?

For ones without Compose reference:
AndroidX: Android, iOS, Desktop (JVM), Desktop (Native)
JB: Everything else, since it most likely will be required as Compose dependency

With Compose dependency:
AndroidX: Android, Desktop (JVM)
JB: Everything else

> For every new such lib, a new couple of lines will need to be added here

Based of count of the libs, I don't really like hardcoding it inside a plugin

Reply:
I see your concern. Let me clarify the purpose and the problem.

The problem this function (with 2 hardcoded lines) solves:

Several compose libraries depend on project(":annotation:annotation") and on (":collection:collection"). 
Without this function klib manifest contains a dependency on `org.jetbrains.compose.annotation-internal` and `org.jetbrains.compose.collection-internal` which can't be resolved because we don't publish ios,macos targets for them (we reference androidx). 

This function will need to substitute only those libraries, which:
- published by androidx for k/native targets
- AND used as dependencies by other compose libraries 

For now, I can't predict how many such cases we'll have.  

We need some way to substitute the project dependency by a module dependency only for certain cases.  Depending on how common those cases become, we can adjust the approach. For now, I can't see a solution without explicitly specifying (hardcode) the libs we need to substitute.

If your concern is only about the place where it's hardcoded, then we'll refactor it separately (to not use Compose specific plugin).
___

As for the libraries that you mentioned.
These two: 
https://maven.google.com/web/index.html?q=lifecy#androidx.lifecycle
https://maven.google.com/web/index.html?q=savedstate#androidx.savedstate
seem to be published only for android, so on our side they won't be different from any compose library (e.g. runtime) we have now.

https://maven.google.com/web/index.html?q=paging#androidx.paging - has ios,macos,linuxX64,jvm,android targets. Unless this library is used within our project as a 'project' dependency of a library (for demo it's not a problem), no hardcode will be necessary.

Reply:
> For now, I can't see a solution without explicitly specifying (hardcode) the libs we need to substitute.

Sounds like another flag in project's gradle properties

> seem to be published only for android

KMP support is just merged there, so the changes here are expected in the next release

Reply:
> KMP support is just merged there, so the changes here are expected in the next release
in 1.7 ?

Reply:
> in 1.7

It's not related to Compose release cycle